### ISSUE TITLE
req: watermark videos

### ISSUE THREAD

**Issue by chovy** [mayeaux/nodetube#133]

**Comment by DontUseGithub** [comment]
You mean for embeds? Hopefully you don't wanna actually patch each video to apply the watermark so that it remains on the video itself forever?

**Comment by chovy** [comment]
No I want to actually add a watermark to the video with ffmpeg. I know how to do it.

**Comment by DontUseGithub** [comment]
So if I'd upload a video I own on nodetube, then let's say I delete or lose the original video, and I wanna download it from a nodetube instance that still has it, will it be watermarked forever? If so then this feels just wrong. Not even PeerTube or Youtube is doing it.

Will this be optional? Will this be turned off by default?

**Comment by mayeaux** [comment]
I can promise it will never be on by default, and I personally will never have it on any of my instances, but I think it's still a cool feature. I have to refactor/reorganize the backend logic for uploading, at that point to add watermarking as an optional ffmpeg step would be pretty trivial.

**Comment by mayeaux** [comment]
That said I will probably never work on it because it's not a priority to me but once the backend refactor is done I'll add on this thread where it the watermark logic should be added and how it can be done with ffmpeg.
